cfme/tests/networks/test_sdn_inventory_collection.py

```diff
@@ -1,30 +1,74 @@
-
-    provider.delete_if_exists(cancel=False)
-    provider.wait_for_delete()
+def test_sdn_api_inventory_networks(provider, appliance):
+    prov_networks = sorted(provider.mgmt.list_network())
+    cfme_networks = sorted([nt.name for nt in appliance.collections.cloud_networks.all()])
+
+    if provider.one_of(EC2Provider):
+        # Ec2 API returns only the Id of the networks, so the alternative is to count the networks
+        # instead of checking there names. Less accurate, but Better than nothing...
+        assert len(cfme_networks) == len(prov_networks), 'There is NOT the same amount of networks'
+        'in CFME than on EC2: {prov} {cfme}'.format(prov=prov_networks, cfme=cfme_networks)
+    else:
+        assert cfme_networks == prov_networks, 'Prov networks list: {networks} different from '
+        'cfme list: {cfme}'.format(networks=prov_networks, cfme=cfme_networks)
+
+
+def test_sdn_api_inventory_subnets(provider, appliance):
+    prov_subnets = []
+    cfme_subnets = [sb.name for sb in appliance.collections.network_subnets.all()]
+    if provider.one_of(AzureProvider):
+        # Because azure is the only provider which return a dict. The same might be done in
+        # Wrapanapi for all providers.
+        for sbn in provider.mgmt.list_subnet().values():
+            prov_subnets.extend(sbn)
+    else:
+        prov_subnets = provider.mgmt.list_subnet()
+
+    assert sorted(cfme_subnets) == sorted(prov_subnets), 'Prov subnets list: {sub} '
+    'different from cfme list: {cfme}'.format(sub=prov_subnets, cfme=cfme_subnets)
+
+
+@pytest.mark.meta(blockers=[BZ(1550605, forced_streams=["5.9", "5.8"],
+                  unblock=lambda provider: not provider.one_of(AzureProvider)),
+                            BZ(1543938, forced_streams=["5.9", "5.8"],
+                               unblock=lambda provider: not provider.one_of(GCEProvider))])
+@pytest.mark.uncollectif(lambda provider, appliance: provider.one_of(EC2Provider) and
+                         appliance.version < "5.9", reason='RFE for 5.9')
+def test_sdn_api_inventory_routers(provider, appliance):
+    prov_routers = sorted(provider.mgmt.list_router())
+    cfme_routers = sorted([rt.name for rt in appliance.collections.network_routers.all()])
+
+    assert cfme_routers == prov_routers, 'Prov routers list: {router} different from cfme list: '
+    '{cfme}'.format(router=prov_routers, cfme=cfme_routers)
+
+
+@pytest.mark.uncollectif(lambda provider: provider.one_of(GCEProvider),
+                         reason='Provider API not provide security groups')
+def test_sdn_api_inventory_security_groups(provider, appliance):
+    prov_sec_gp = sorted(provider.mgmt.list_security_group())
+    cfme_sec_gp = sorted([sec.name for sec in appliance.collections.network_security_groups.all()])
+
+    assert prov_sec_gp == cfme_sec_gp, 'Prov security groups list: {sec} different from '
+    'cfme list: {cfme}'.format(sec=prov_sec_gp, cfme=cfme_sec_gp)
+
+
```

Comment:
**REQUIRED:** Please use the provider marker instead specifically for this test

Reply:
changed